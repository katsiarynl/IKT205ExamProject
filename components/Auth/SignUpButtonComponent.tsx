import { View, Text, TouchableOpacity } from "react-native";
import React from "react";
import { signInStyle } from "../../styles/signIn";
import { useNavigation } from "@react-navigation/native";
import { NavigationSignOut } from "../../types/navigationTypes";

export default function SignUpButtonComponent() {
  const navigation = useNavigation<NavigationSignOut>();
  return (
    <View style={signInStyle.singUpContainer}>
      <Text style={signInStyle.textSingUp}>Don't have an account? </Text>
      <TouchableOpacity
        testID="SignUpIfNotRegistered"
        onPress={() => navigation.navigate("SignUp")}
      >
        <Text style={{ fontWeight: "bold", color: "#7E3B14", fontSize: 19 }}>
          Sing Up
        </Text>
      </TouchableOpacity>
    </View>
  );
}
