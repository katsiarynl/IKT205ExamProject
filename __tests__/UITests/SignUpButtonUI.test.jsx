import SignUpButtonComponent from "../../components/Auth/SignUpButtonComponent";

import { create } from "react-test-renderer";

const mockedNavigate = jest.fn();
//https://stackoverflow.com/questions/61781274/how-to-mock-usenavigation-hook-in-react-navigation-5-0-for-jest-test
jest.mock("@react-navigation/native", () => ({
  useNavigation: () => ({ navigate: mockedNavigate }),
}));
const tree = create(<SignUpButtonComponent />);

test("snapshot", () => {
  expect(tree).toMatchSnapshot();
});
