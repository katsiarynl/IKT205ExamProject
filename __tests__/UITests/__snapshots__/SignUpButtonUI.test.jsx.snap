// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "marginTop": 20,
    }
  }
>
  <Text
    style={
      {
        "color": "grey",
        "fontSize": 16,
        "fontWeight": "600",
      }
    }
  >
    Don't have an account? 
  </Text>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
    testID="SignUpIfNotRegistered"
  >
    <Text
      style={
        {
          "color": "#7E3B14",
          "fontSize": 19,
          "fontWeight": "bold",
        }
      }
    >
      Sing Up
    </Text>
  </View>
</View>
`;
